src/styles/utils/_variables.scss

```diff
@@ -0,0 +1,20 @@
+$gap: 1px;
+$first-day: 1;
+$number-of-days-month: 31;
+$number-of-days-week: 7;
+$min-last-days-month: 28;
+$max-last-days-month: 31;
+$width: 100px;
+$width-week:
+  $width * $number-of-days-week +
+  ($gap * $number-of-days-week) - 1;
+
+$days: (
+  "sun" $width-week - $width,
+  "sat" $width-week - $width * 2 - $gap,
+  "fri" $width-week - $width * 3 - $gap * 2,
+  "thu" $width-week - $width * 4 - $gap * 3,
+  "wed" $width-week - $width * 5 - $gap * 4,
+  "tue" $width-week - $width * 6 - $gap * 5,
+  "mon" $width-week - $width * 7 - $gap * 6,
```

Comment:
expressions can be replaced with one formula.

Reply:
Can you be a bit more specific? I can't think of a way to do that without changing ~20% of the project

Reply:
`$width-week - $width * $index - $gap * ($index - 1)` in the loop.

Reply:
And replace it here with 1, 2, 3 etc.

Reply:
I'm sorry, I don't understand how I'm supposed to use that in the @each loop, or do you mean to make a nested loop?

Reply:
You create the collection witn words and indexes.
`"mon" 0, "tue" 1, ... `

And then in the `@each` loop:

```
@each $day, $index in $weekdays {
    &--start-day--#{$day} > :first-child {
      margin-left: ($box-size + $gap) * $index;
    }
}
```

Reply:
Oh that's what you meant, I completely misunderstood you and was halfway through making a @for @each loop 😅

Reply:
Nested loops are evil 👿

Reply:
Because of CSS or something else?

Reply:
Overall. In all languages. You can read about algorithm complexity.